### PR TITLE
Bring search improvements out from behind debug menu

### PR DESCRIPTION
**Pull request by scottmitchell**
### Purpose

This PR updates finalizes improvements to search made in [DYN-1326](https://jira.autodesk.com/browse/DYN-1326) and [DYN-1324](https://jira.autodesk.com/browse/DYN-1326). These improvements were initially placed behind the debug menu. This PR brings both improvements out from behind the debug menu and removes them from the debug menu.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@ColinDayOrg 
### FYIs
@Racel

### CODE REVIEW

File: src/DynamoCore/Search/SearchDictionary.cs [hunk 1]
```diff
@@ -372,11 +359,10 @@ internal IEnumerable<V> Search(string query, int minResultsForTolerantSearch = 0
...
             var searchResults = orderedSearchDict.Select(x => x.Key);
-            if (TruncateSearchResults)
-            {
-                searchResults = searchResults.Take(20);
-            }
 
+            // return only the top 20 search results
+            searchResults = searchResults.Take(20);
+            
 #if DEBUG
```
Comment (scottmitchell) [inline src/DynamoCore/Search/SearchDictionary.cs]: @ColinDayOrg I'm assuming we want to leave the logging work you did?

**Comment by Racel** [comment]
just doing another pass at all of this. i think we should change the number of search results to 30. will share the results in a bit. still going through the data

**Comment by scottmitchell** [comment]
@Racel Just updated to return top 30 instead of top 20.

File: src/DynamoCore/Search/SearchDictionary.cs [hunk 2]
```diff
@@ -372,11 +359,10 @@ internal IEnumerable<V> Search(string query, int minResultsForTolerantSearch = 0
             var orderedSearchDict = searchDict.OrderByDescending(x => x.Value);
 
             var searchResults = orderedSearchDict.Select(x => x.Key);
-            if (TruncateSearchResults)
-            {
-                searchResults = searchResults.Take(20);
-            }
 
+            // return only the top 30 search results
```
Comment (ColinDayOrg) [inline src/DynamoCore/Search/SearchDictionary.cs]: Not that I'm against it, but why the change from 20 to 30 here?

File: src/DynamoCore/Search/SearchDictionary.cs (hunk 1 above)
Comment (ColinDayOrg) [reply src/DynamoCore/Search/SearchDictionary.cs]: I think so, it's not a lot of logging output and the output could be useful.

File: src/DynamoCore/Search/SearchDictionary.cs (hunk 2 above)
Comment (ColinDayOrg) [reply src/DynamoCore/Search/SearchDictionary.cs]: Just saw the conversation in the main comments, it didn't show up in the file comparison view. This is fine with me.

**Comment by ColinDayOrg** [comment]
LGTM :shipit:

File: src/DynamoCore/Search/SearchDictionary.cs (hunk 1 above)
Comment (ColinDayOrg) [reply src/DynamoCore/Search/SearchDictionary.cs]: Odd that file comments don't show up here always and vice versa, so repeating here, it's not a lot of  logging output and it could be useful (and is only under debug anyways) so I would say yes.

**Comment by Racel** [comment]
Hey guys- so, I did a thorough review of the work against the search analysis results. And, the results look like that people are likely to just type in a new search after not finding it in the top 10. Or on the first page, can we please just set the number back to 20. I think our instincts were correct on this one, and it looks like the data backs that up. See the Jira task for more details.
